Document the ability to hide cells based on cell tags

AartGoossens opened the pull request:
This PR documents the ability to hide code and output cells based on cell tags.
Related issue: https://github.com/voila-dashboards/voila/issues/171

AartGoossens commented:
I also changed some headings to proper markdown headings so you can directly link to subsections in the README.

jtpio commented:
Many thanks @AartGoossens for this.

I wonder if these instructions could also go to the documentation, under the customizing section? https://voila.readthedocs.io/en/latest/customize.html

jtpio commented:
And this should also fix https://github.com/voila-dashboards/voila/issues/171.

AartGoossens commented:
I updated the PR: moved the documentation to `docs/source/customize.rst` (which is indeed a much better place).

Let me know what else I can do before this can be merged.

jtpio commented on docs/source/customize.rst:
```diff
@@ -232,3 +232,20 @@ Python this would be a call to `IPython.display.display`.
 script files. After installing jupytext, voila will see script files as if they
 are notebooks, and requires no extra configuration.
+
+Hiding ouput and code cells based on cell tags
```
```suggestion
Hiding output and code cells based on cell tags
```

jtpio commented on docs/source/customize.rst:
```diff
@@ -232,3 +232,20 @@ Python this would be a call to `IPython.display.display`.
+Hiding ouput and code cells based on cell tags
+==============================================
+
+Voila uses `nbconvert <https://github.com/jupyter/nbconvert>`_ under the hood to render the notebooks so we can benefit from some of it's advanced functionality to hide code and output cells based on cell tags.
```
```suggestion
Voila uses `nbconvert <https://github.com/jupyter/nbconvert>`_ under the hood to render the notebooks so we can benefit from some of its advanced functionality to hide code and output cells based on cell tags.
```

jtpio commented on docs/source/customize.rst:
```diff
@@ -232,3 +232,20 @@ Python this would be a call to `IPython.display.display`.
+
+    voila --TagRemovePreprocessor.remove_cell_tags='{"hide"}' your_notebook.ipynb
+
+You can use any tag you want to use this functionality but be sure to use the same tag name in the Voila command.
```
Maybe a bit more concise?

```suggestion
You can use any tag you want but be sure to use the same tag name in the Voila command.
```

jtpio commented:
Thanks again @AartGoossens, left a couple of inline comments (minor typos).

AartGoossens commented:
@jtpio Thanks for taking a good look and noticing the typos. :)
I fixed them and pushed my changes.

maartenbreddels commented:
good catch @SylvainCorlay :)
